### PR TITLE
Cleanup and fixes

### PR DESCRIPTION
**Pull request by skonto**
- removes unused code
- handles some nil issues
- handles some deprecation issues

### CODE REVIEW

File: test/clients.go [hunk 1]
```diff
@@ -137,9 +137,6 @@ func NewClients(kubeconfig string) (*Clients, error) {
 	}
 
 	clients.ConsoleCLIDownload = consolev1.NewForConfigOrDie(cfg).ConsoleCLIDownloads()
-	if err != nil {
```
Comment (skonto) [inline test/clients.go:140]: This is a relic of the past.

File: test/service.go [hunk 2]
```diff
@@ -175,6 +133,9 @@ func WaitForOperatorDepsDeleted(ctx *Context) error {
 
 	waitErr := wait.PollImmediate(Interval, Timeout, func() (bool, error) {
 		existingDeployments, err := ctx.Clients.Kube.AppsV1().Deployments(OperatorsNamespace).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return true, err
+		}
```
Comment (skonto) [inline test/service.go:138]: We may get a nil pointer when iterating.

File: test/service.go [hunk 3]
```diff
@@ -72,34 +70,6 @@ func CreateService(ctx *Context, name, namespace, image string) (*servingv1.Serv
 	return service, nil
 }
 
-func WaitForControllerEnvironment(ctx *Context, ns, envName, envValue string) error {
```
Comment (skonto) [inline test/service.go:75]: Not used.

File: knative-operator/pkg/controller/knativeserving/consoleclidownload/consoleclidownload.go [hunk 4]
```diff
@@ -160,20 +160,19 @@ func deleteDeprecatedResources(instance *servingv1alpha1.KnativeServing, apiclie
 
 // makeKnService makes Knative Service object and its SPEC from provided image parameter
 func makeKnService(image string, instance *servingv1alpha1.KnativeServing) *servingv1.Service {
+	if instance == nil {
+		return nil
+	}
 	// OwnerReference is not used to handle ksvc cleanup due to race condition with control-plane deletion.
 	// In such a case route's finalizer blocks clean removal of resources.
-	anno := make(map[string]string)
-	if instance != nil {
```
Comment (skonto) [inline knative-operator/pkg/controller/knativeserving/consoleclidownload/consoleclidownload.go:166]: Does not make sense to test for nil here and not bellow.

**Comment by nak3** [comment]
/lgtm

Thank you!

File: serving/ingress/pkg/reconciler/ingress/ingress_test.go [hunk 5]
```diff
@@ -74,7 +74,7 @@ func TestKourierReconcile(t *testing.T) {
 		WantDeletes: []clientgotesting.DeleteActionImpl{{
 			ActionImpl: clientgotesting.ActionImpl{
 				Namespace: ingressNamespace,
-				Resource:  routev1.SchemeGroupVersion.WithResource("routes"),
+				Resource:  routev1.GroupVersion.WithResource("routes"),
```
Comment (markusthoemmes) [inline serving/ingress/pkg/reconciler/ingress/ingress_test.go:77]: Curious: Why is this better than the other?

**Review by markusthoemmes — APPROVED** [review]

File: serving/ingress/pkg/reconciler/ingress/ingress_test.go (hunk 5 above)
Comment (skonto) [reply serving/ingress/pkg/reconciler/ingress/ingress_test.go:77]: @markusthoemmes it was deprecated.

**Pull request merged**